### PR TITLE
Add support for RISC-V

### PR DESCRIPTION
**Pull request by chazapis**
### 1. Why is this pull request needed and what does it do?

This PR adds RISC-V support, by generating a RISC-V binary and container image.

### 2. Which issues (if any) are related?

No relevant issues found.

### 3. Which documentation changes (if any) need to be made?

None.

### 4. Does this introduce a backward incompatible change or deprecation?

No.

### Notes

By adding the architecture to `Makefile.release`, I was able to easily generate a binary for `riscv64`.

The container image generation was a bit trickier, as the base images used do not currently support RISC-V (`debian:stable-slim` and `gcr.io/distroless/static-debian11:nonroot`). Debian supports RISC-V in the "unstable" branch and distroless has [deferred releasing RISC-V images until the unstable branch is moved into stable](https://github.com/GoogleContainerTools/distroless/issues/1269) (estimated in 2025). ~Until then, the cleanest solution I could come up with is to use `Dockerfile.riscv64` for RISC-V with a simple and generic change in `Makefile.docker` to select the architecture-specific `Dockerfile` if available.~ Edit: For RISC-V we use unstable-derived base images, until there are RISC-V variants of the images used for all other architectures.

The resulting container has been tested in QEMU with a [RISC-V build of K3s](https://github.com/k3s-io/k3s/pull/7778).

### CODE REVIEW

File: Dockerfile.riscv64 [hunk 1]
```diff
@@ -0,0 +1,18 @@
+ARG DEBIAN_IMAGE=debian:unstable-slim
```
Comment (SuperQ) [inline Dockerfile.riscv64]: There's no need to have a separate Dockerfile for this. We don't use any architecture-specific pieces in the Docker build.

**Review by SuperQ — CHANGES_REQUESTED** [review]

**Comment by SuperQ** [comment]
Secondary, we don't need a whole new Dockerfile, we can simply adjust the image with a `DEBIAN_IMAGE` env var.

File: Dockerfile.riscv64 (hunk 1 above)
Comment (chazapis) [reply Dockerfile.riscv64]: This is not the only reason for the separate `Dockerfile`. There is also the problem of not having a distroless image with a `nonroot` user. Yes, the Debian image could be adjusted by setting the variable, but are there any better ideas on what to do with the missing distroless image?

Comment (chazapis) [reply Dockerfile.riscv64:1]: For example, using Alpine for the final image would be a solution, as Alpine supports RISC-V, but this is a rather big change (moving to Alpine for all architectures).

Comment (chazapis) [reply Dockerfile.riscv64]: Ok, I just found into [pre-made images of distroless for RISC-V](https://github.com/go-riscv/distroless/pkgs/container/distroless%2Fstatic-unstable). Somehow I missed that. I will try with the existing `Dockerfile`.

**Comment by chazapis** [comment]
> Secondary, we don't need a whole new Dockerfile, we can simply adjust the image with a `DEBIAN_IMAGE` env var.

You are right. I uploaded a new version that just sets the base images to "unstable" variants with RISC-V support. I have tested this in a QEMU setup and it looks like it is working fine.

**Review by SuperQ — APPROVED** [review]
Nice, thanks!

**Pull request merged**